### PR TITLE
LXD 5.0.3 (5.0-candidate)

### PR DESCRIPTION
**Pull request by tomponline**
Brings 5.0-candidate into sync with 5.0-edge

### CODE REVIEW

File: snapcraft.yaml [hunk 1]
```diff
@@ -1541,19 +1570,48 @@ parts:
...
+        -exec strip -s {} +
+
+      # Strip all versions of zfs utils
+      for v in "${SNAPCRAFT_PRIME}"/zfs-*; do
+        [ -d "${v}" ] || continue
+        find "${v}/" -type f -exec strip -s {} +
       done
 
-      [ -e "${SNAPCRAFT_PRIME}/criu/criu" ] && strip -s ${SNAPCRAFT_PRIME}/criu/criu
+      # Strip libraries (excluding python3 scripts)
```
Comment (simondeziel) [inline snapcraft.yaml:1609]: It seems the criu stripping bits is missing here.

File: snapcraft/wrappers/editor [hunk 2]
```diff
@@ -15,6 +15,11 @@ run_cmd() {
     exec unshare --kill-child -U -m -p -r -f -R "/var/lib/snapd/hostfs/" "/bin/sh" -c "mount -t proc proc /proc 2>/dev/null || true; exec \"${CMD}\" \"$@\""
 }
 
+USERNS=1
+[ -e /proc/sys/kernel/unprivileged_userns_clone ] && grep -qxF 0 /proc/sys/kernel/unprivileged_userns_clone && USERNS=0
+[ -e /proc/sys/kernel/apparmor_restrict_unprivileged_userns ] && grep -qxF 1 /proc/sys/kernel/apparmor_restrict_unprivileged_userns && USERNS=0
+[ -e /proc/sys/kernel/apparmor_restrict_unprivileged_unconfined ] && grep -qxF 1 /proc/sys/kernel/apparmor_restrict_unprivileged_unconfined && USERNS=0
+
```
Comment (simondeziel) [inline snapcraft/wrappers/editor:22]: Should we drop those (commit be91e3426d120430ab7d42ba4c579c2784362787)

File: snapcraft/wrappers/remote-viewer [hunk 3]
```diff
@@ -14,6 +14,11 @@ run_cmd() {
     exec unshare -U -r chroot "/var/lib/snapd/hostfs/" "${CMD}" "$@"
 }
 
+USERNS=1
+[ -e /proc/sys/kernel/unprivileged_userns_clone ] && grep -qxF 0 /proc/sys/kernel/unprivileged_userns_clone && USERNS=0
+[ -e /proc/sys/kernel/apparmor_restrict_unprivileged_userns ] && grep -qxF 1 /proc/sys/kernel/apparmor_restrict_unprivileged_userns && USERNS=0
+[ -e /proc/sys/kernel/apparmor_restrict_unprivileged_unconfined ] && grep -qxF 1 /proc/sys/kernel/apparmor_restrict_unprivileged_unconfined && USERNS=0
+
```
Comment (simondeziel) [inline snapcraft/wrappers/remote-viewer:21]: Same for those (commit e387b3a130bbe2dca59336349ac49dabc3f2a765)?

File: snapcraft.yaml (hunk 1 above)
Comment (tomponline) [reply snapcraft.yaml:1609]: Which commit adds them from 5.0-edge?

File: snapcraft/wrappers/editor (hunk 2 above)
Comment (tomponline) [reply snapcraft/wrappers/editor:22]: They appear not to be dropped in 5.0-edge, so they should not be dropped no?

**Comment by tomponline** [comment]
@simondeziel in general ive brought 5.0-candidate inline with 5.0-edge.

Do you think there are things in latest-* that should be in 5.0-edge (and by extension 5.0-candidate)?

File: snapcraft.yaml (hunk 1 above)
Comment (simondeziel) [reply snapcraft.yaml:1609]: Ah, you've spot my mistake in 5.0-edge where commit f59b44b273fb83286e3857a140a108682493aacf removed the criu stripping.

In latest-edge, it was re-introduced with commit 27ed3277ea7e1b936b2db74af5e34fc5fea94727 and its fixup ce50982d41818478746e4d239f4e402c9f7e84f9.

Comment (tomponline) [reply snapcraft.yaml:1609]: @simondeziel im EOD now, please can you include any commits you think should be in 5.0-candidate by opening a PR for 5.0-edge and then i'll merge and cherry-pick into this PR tomorrow. cheers

Comment (tomponline) [reply snapcraft.yaml:1609]: Also you can include in your PR to 5.0-edge restoring this line https://chat.canonical.com/canonical/pl/eyigh3kxhi8h8gzbugkd1wm5ke

File: snapcraft/wrappers/editor (hunk 2 above)
Comment (simondeziel) [reply snapcraft/wrappers/editor:22]: Right, it probably doesn't hurt to keep them in considering those files are now readable on all updated kernels.

File: snapcraft.yaml (hunk 1 above)
Comment (simondeziel) [reply snapcraft.yaml:1609]: Sure thing, thanks for catching this!

**Comment by simondeziel** [comment]
> Do you think there are things in latest-* that should be in 5.0-edge (and by extension 5.0-candidate)?

I intend to have the criu stripping and ZFS-2.2 build-attributes sent through a PR targeting 5.0-edge. That's the only thing I can think of.

**Comment by tomponline** [comment]
Thanks

File: snapcraft.yaml (hunk 1 above)
Comment (simondeziel) [reply snapcraft.yaml:1609]: https://github.com/canonical/lxd-pkg-snap/pull/293

**Comment by tomponline** [comment]
Included https://github.com/canonical/lxd-pkg-snap/pull/293 and https://github.com/canonical/lxd-pkg-snap/pull/295

**Comment by tomponline** [comment]
@simondeziel ready for review.

**Review by simondeziel — APPROVED** [review]

**Pull request merged**